### PR TITLE
match authrealm ldap url to Openshift's ldap url 

### PR DESCRIPTION
**Pull request by leena-jawale**
issue: https://github.com/stolostron/backlog/issues/22283

/hold

### CODE REVIEW

File: controllers/authrealm/dex.go
Comment (itdove) [inline controllers/authrealm/dex.go:413]: could you use a switch statement here ?

File: controllers/authrealm/dex.go
Comment (itdove) [inline controllers/authrealm/dex.go:449]: same here (switch) ?

File: controllers/authrealm/dex.go
Comment (chrisahl) [inline controllers/authrealm/dex.go:444]: Is this logic correct or backwards?  Wouldn't ldaps want TLS?

File: go.mod
Comment (chrisahl) [inline go.mod:14]: Assume we need to wait for idp-client-api change to get merged so you can pull it in with a change to go mod before merging this PR

Comment (leena-jawale) [reply go.mod:14]: yeah thats right

File: controllers/authrealm/dex.go
Comment (leena-jawale) [reply controllers/authrealm/dex.go]: Its other way round. if there ldaps its using TLS connection to server. If there ldap, then wwe have to providee startTLS =true, to upgrade the connection to TLS

Comment (leena-jawale) [reply controllers/authrealm/dex.go]: Screen Shot 2022-05-18 at 8 25 17 AM

File: controllers/authrealm/dex.go
Comment (robinbobbitt) [inline controllers/authrealm/dex.go:411]: The OpenShift docs say they only support "one" or "sub", as does Dex. See https://github.com/dexidp/dex/blob/9d3471e39b24c11350b8c911fc6e736f0be7c785/connector/ldap/ldap.go#L174

What does OpenShift do if you put a scope of base? Seems we should not allow it.

File: controllers/authrealm/dex.go
Comment (robinbobbitt) [inline controllers/authrealm/dex.go:433]: Let's update the unit tests to be sure we're confirming these values are being mapped from the URL.

**Review by robinbobbitt — CHANGES_REQUESTED** [review]
Great improvements! I've added a few comments and requests. Thanks!

File: controllers/authrealm/dex.go
Comment (leena-jawale) [reply controllers/authrealm/dex.go:411]: I'll remove base.

File: controllers/authrealm/dex.go
Comment (leena-jawale) [reply controllers/authrealm/dex.go:431]: updated

File: controllers/authrealm/dex.go
Comment (leena-jawale) [reply controllers/authrealm/dex.go:402]: added webhook check in same PR

File: controllers/authrealm/dex.go
Comment (robinbobbitt) [inline controllers/authrealm/dex.go:447]: This configuration of insecure connection, not upgraded leaks passwords, and per Dex docs may not be supported in the future. Shall we perhaps not support this option? We could update our webhook to never allow `insecure: true` with our operator.

File: controllers/authrealm/suite_test.go
Comment (robinbobbitt) [inline controllers/authrealm/suite_test.go:733]: Better to use `EmailAttr` here?

File: controllers/authrealm/suite_test.go
Comment (robinbobbitt) [inline controllers/authrealm/suite_test.go:564]: Shall we build this URL from Filter, BaseDN etc to be more clear how these all tie together? Same comment for ldaps test.

File: controllers/authrealm/suite_test.go
Comment (robinbobbitt) [inline controllers/authrealm/suite_test.go:637]: Shall we check insecureNoSSL too? Same for ldaps test?

File: controllers/authrealm/dex.go
Comment (leena-jawale) [reply controllers/authrealm/dex.go:447]: Yeah I agree with not allowing `insecure: true`. I will update the webhook check to not allow `insecure: true`

**Pull request merged**
